src/dmd/expressionsem.d

```diff
@@ -1125,6 +1132,12 @@ private Expression resolvePropertiesX(Scope* sc, Expression e1, Expression e2 =
     Dsymbol s;
     Objects* tiargs;
     Type tthis;
+    CommaExp ce1;
+    if (e1.op == TOK.comma)
```

Comment:
do this in a function which you call before return, than you won't need to goto.